File: fixtures/scale/objc/large/Families/Caesar/CaesarDescriptor.m
#import "CaesarDescriptor.h"
#import "CaesarCipher.h"
#import "CaesarVectors.h"

@implementation CaesarDescriptor

- (NSString *)family {
    return @"caesar";
}

- (NSString *)suite {
    return @"classical";
}

- (id<CLCipher>)cipher {
    return [[CaesarCipher alloc] init];
}

- (NSArray *)vectors {
    return [CaesarVectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Caesar/CaesarDescriptor.php
<?php

namespace CipherLab\Families\Caesar;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together CaesarCipher and
// CaesarVectors under the "classical" suite.
class CaesarDescriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'caesar';
    }

    public function suite(): string
    {
        return 'classical';
    }

    public function cipher(): Cipher
    {
        return new CaesarCipher();
    }

    public function vectors(): array
    {
        return CaesarVectors::all();
    }
}
